### PR TITLE
feat(gateway): add enterprise control-plane integration

### PR DESCRIPTION
**Pull request by desertsurge**
## Summary

Adds a token-authenticated enterprise integration surface so an external AIP control plane can synchronize tenant topology and manage per-principal model access keys while TokenHub remains the runtime authorization and model-discovery authority.

## Related Issue

N/A

## Changes

- Backend: add idempotent integration event ingestion and reconciliation for tenants, organizations, principals, projects, and workloads.
- Backend: add internal APIs for model/provider/route discovery, scoped model access key creation, listing, reveal, revocation, request logs, and usage reporting.
- Backend: protect AIP-managed projects and keys from normal mutation paths, encrypt revealable key material at rest, and enforce tenant, principal, lifecycle, and request-id constraints.
- Tests: add integration, infrastructure, model access key, request log, usage, and multi-instance coverage.
- Deployment and documentation: add `TOKENHUB_INTEGRATION_TOKEN` to environment examples, Compose definitions, install scripts, startup validation, and English, Simplified Chinese, and Japanese deployment guides.
- Frontend and SDK: no feature-specific changes.

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor or maintenance
- [ ] Documentation
- [x] Deployment or configuration

## Verification

- [ ] Backend: `gofmt` on changed Go files, `go test ./...`, and `go vet ./...`
- [ ] Frontend: `npm run typecheck` and `npm run build`
- [ ] SDK smoke tests against a compatible backend
- [ ] Docker Compose configuration rendered successfully
- [x] Other focused or manual verification described below

Verification details:

- PASS: resolved the merge against current `main`; the branch is 0 commits behind and `git diff --check origin/main...HEAD` passes.
- PASS: formatted the conflict-resolved Go files and reviewed the 25-file PR diff against current `main`.
- PASS: `frontend/node_modules/.bin/tsc.cmd --noEmit` and `npm run build`.
- BLOCKED: `go test ./...` downloaded Go 1.26 successfully but this Windows host has `CGO_ENABLED=0` and no GCC/Clang. Compilation stops at the existing `origin/main` SQLite backup call (`store.go`: `dest.Backup undefined`). `go vet ./...` is blocked by the same environment limitation.
- BASELINE FAILURE: `npm run typecheck` reaches the source-line check and reports three files already unchanged from `origin/main` above the 1,500-line limit: `features/admin/i18n/en.tsx`, `features/admin/i18n/ja.tsx`, and `features/admin/views/provider-editor.tsx`. Direct TypeScript compilation passes.
- NOT RUN: SDK smoke tests require a compatible running backend and provider credentials.
- NOT RUN: Docker Compose rendering because Docker Desktop is not running on this host.

## Compatibility, Security, and Operations

- OpenAI-compatible `/v1` API impact: No intentional contract changes. The new endpoints are under `/api/internal`; AIP-managed keys authenticate through the existing gateway path.
- Security or credential-handling impact: Adds a dedicated `TOKENHUB_INTEGRATION_TOKEN`. Model access secrets are returned only by create/reveal operations, stored encrypted at rest, excluded from public serialization, and constrained by tenant, principal, status, expiry, and idempotency rules.
- Database, environment, or deployment impact: Adds control-plane synchronization tables and additive API key fields through AutoMigrate. Deployments must configure a strong integration token; examples, Compose files, startup scripts, validation, and deployment guides are synchronized.
- Rollout and rollback considerations: Configure and rotate the integration token before enabling AIP traffic. Roll back by stopping AIP calls and deploying the previous binary; additive database columns/tables can remain without affecting existing `/v1` traffic.

## Checklist

- [x] Tests were added or updated for behavior changes, or the reason they are unnecessary is documented.
- [x] No credentials, local `.env` files, databases, backups, or runtime logs are included.
- [x] Environment variable changes are synchronized across examples, Compose, `start.sh`, and deployment documentation where applicable.
- [x] Shared user-facing behavior is documented consistently in English, Simplified Chinese, and Japanese where applicable.
- [x] `data/model-catalog.yaml` remains tracked and catalog changes were reviewed where applicable.
- [x] `git diff --check` passes.

### CODE REVIEW

File: backend/internal/server/store.go [hunk 1]
```diff
@@ -929,6 +944,9 @@ func (s *GormStore) UpdateProject(id string, patch Project) (Project, error) {
 	if err := s.db.First(&project, "id = ?", id).Error; err != nil {
 		return Project{}, notFound(err, "project_not_found", "Project not found")
 	}
+	if err := rejectGatewayManagedProjectMutation(s.db, id); err != nil {
```
Comment (astaxie) [inline backend/internal/server/store.go]: **[P1] 托管项目只读保护仍可通过创建接口绕过。**

这里保护了 Update，Delete 也有相同检查，但 `POST /api/admin/projects` 接受调用方提供的 `Project.ID`，而 `CreateProject` 使用 `OnConflict{UpdateAll: true}` 且没有调用此检查。管理员只要复用一个可见的托管项目 ID，就能覆盖其服务记录，包括团队、成本中心和配额等后续投影不会恢复的字段。建议让 `CreateProject` 返回错误并在 ID 已属于 `GatewayProject` 时拒绝，或禁止创建接口接受已有 ID；同时增加回归测试。该路径已通过最小测试复现。

**[P1] Managed-project read-only protection can still be bypassed through the create endpoint.**

Update is protected here and Delete has the same check, but `POST /api/admin/projects` accepts a caller-supplied `Project.ID`, while `CreateProject` uses `OnConflict{UpdateAll: true}` without this guard. An administrator can reuse a visible managed-project ID and overwrite its serving row, including team, cost-center, and quota fields that later projections do not restore. Please make `CreateProject` return an error when the ID belongs to a `GatewayProject`, or prevent the create endpoint from accepting an existing ID, and add a regression test. This path was reproduced with a minimal test.

File: backend/internal/server/gateway_request_logs.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+		base = base.Where("rl.created_at <= ?", *filter.DateTo)
+	}
+
+	var total int64
+	if err := base.Distinct("rl.id").Count(&total).Error; err != nil {
+		return GatewayRequestLogPage{}, err
+	}
+	items := make([]GatewayRequestLogItem, 0)
+	err := base.
+		Joins("LEFT JOIN usage_records AS ur ON ur.request_id = rl.request_id").
```
Comment (astaxie) [inline backend/internal/server/gateway_request_logs.go]: **[P1] 仅按 `request_id` 关联会破坏租户隔离。**

`rl` 已通过租户的 API Key 限定范围，但这里的 usage 关联没有约束 `ur.api_key_id` 或租户。数据库也没有保证 Request ID 全局唯一，因此相同 Request ID 的另一租户 usage 会被汇总进当前租户的日志。已用两个租户的最小数据集复现。建议至少关联 `ur.request_id = rl.request_id AND ur.api_key_id = rl.api_key_id`，并增加重复 Request ID 的跨租户测试。

**[P1] Joining only on `request_id` breaks tenant isolation.**

`rl` is scoped through the tenant's API key, but this usage join does not constrain `ur.api_key_id` or tenant ownership. The database does not enforce global Request ID uniqueness, so usage from another tenant with the same Request ID is summed into the current tenant's log. This was reproduced with a minimal two-tenant dataset. Please join on at least `ur.request_id = rl.request_id AND ur.api_key_id = rl.api_key_id` and add a cross-tenant duplicate-Request-ID test.

File: backend/internal/server/gateway_model_access_keys.go [hunk 3]
```diff
@@ -0,0 +1,703 @@
...
+	key, err := s.store.RevokeGatewayModelAccessKey(parts[0], input.TenantExternalID, input.PrincipalType, input.PrincipalID, input.Reason, input.RequestedBy)
+	if err != nil {
+		writeError(w, r, err)
+		return
+	}
+	writeJSON(w, http.StatusOK, map[string]any{"data": key})
+}
+
+func (s *GormStore) CreateGatewayModelAccessKey(input GatewayModelAccessKeyCreateInput) (GatewayModelAccessKeyCreateResult, error) {
+	normalized, digest, err := normalizeGatewayModelAccessKeyCreateInput(input)
```
Comment (astaxie) [inline backend/internal/server/gateway_model_access_keys.go:172]: **[P2] 过期后同一个 `request_id` 无法幂等重放。**

这里先执行完整规范化和“过期时间必须晚于当前时间”校验，之后才查询已有 Control Request。首次创建成功后，如果控制面在 Key 到期后重放完全相同的请求，会在查找已有记录之前返回 400，而不是返回已有 Key 的幂等响应。建议先从稳定字段解析 tenant/request identity 并查找已有请求，再对真正的新请求执行时间相关校验；请增加跨过期时间边界的重放测试。

**[P2] The same `request_id` is no longer idempotently replayable after expiry.**

This performs full normalization, including the requirement that expiration remain in the future, before looking up an existing control request. After the first create succeeds, replaying the identical request after the key expires returns 400 before the existing record is found instead of returning an idempotent response. Please derive the tenant/request identity and check for an existing request before time-relative validation of a new request, and add a replay test that crosses the expiration boundary.

File: backend/internal/server/gateway_model_access_keys.go [hunk 4]
```diff
@@ -0,0 +1,703 @@
...
+	return StatusActive
+}
+
+func validateGatewayModelAccessKeyEffectiveScope(db *gorm.DB, key APIKey) error {
+	var tenant GatewayTenant
+	if err := db.First(&tenant, "external_tenant_id = ? AND status = ?", key.TenantExternalID, StatusActive).Error; err != nil {
+		return ErrAPIKeyDisabled
+	}
+	var project GatewayProject
+	if err := db.First(&project, "id = ? AND tenant_id = ? AND status = ?", key.ProjectID, tenant.ID, StatusActive).Error; err != nil {
```
Comment (astaxie) [inline backend/internal/server/gateway_model_access_keys.go:686]: **[P2] 组织生命周期没有参与运行时授权。**

有效范围只检查 tenant、project 和 principal/workload。`organization.disabled` 或 `organization.deleted` 只更新组织投影，因此组织下项目的现有 Key 仍保持可用。如果组织状态属于授权生命周期，请在这里校验项目关联组织及必要的成员关系；如果组织只用于归因，请在 API 契约和测试中明确这一点，避免控制面误以为停用组织会阻断访问。

**[P2] Organization lifecycle is not enforced by runtime authorization.**

The effective scope checks only the tenant, project, and principal/workload. `organization.disabled` and `organization.deleted` only update the organization projection, so existing keys for projects under that organization remain usable. If organization status is an authorization boundary, validate the project's organization and any required membership here. If organizations are attribution-only, document and test that explicitly so the control plane does not assume disabling an organization blocks access.

File: backend/internal/server/gateway_integration.go [hunk 5]
```diff
@@ -0,0 +1,750 @@
...
+		},
+	}
+	if err := s.db.Model(&IntegrationInbox{}).Where("tenant_id = ?", tenantID).Count(&summary.ReceivedEvents).Error; err != nil {
+		return GatewayIntegrationReconciliationSummary{}, err
+	}
+	var groups []struct {
+		AggregateType string
+		EventCount    int64
+	}
+	if err := s.db.Model(&IntegrationInbox{}).Where("tenant_id = ?", tenantID).
```
Comment (astaxie) [inline backend/internal/server/gateway_integration.go:327]: **[P2] 当前 reconciliation 只能比较事件数量，不能核对拓扑状态。**

响应没有返回实体 ID、当前版本、状态或删除标记，因此 AIP 最多只能发现事件总量不一致，无法定位缺失、落后或状态分歧的投影。这与 PR 所述的 tenant/org/principal/project/workload reconciliation 存在落差。建议返回可分页的投影摘要、稳定摘要哈希或每类版本水位；如果这里只计划做 inbox 水位检查，应相应调整接口命名和文档。

**[P2] The current reconciliation endpoint can compare event counts but cannot reconcile topology state.**

The response contains no entity IDs, current versions, statuses, or deletion markers. AIP can at most detect a difference in event volume, but cannot identify a missing, stale, or divergent projection. That falls short of the tenant/org/principal/project/workload reconciliation described by the PR. Please expose pageable projection summaries, stable digests, or per-aggregate version watermarks; if this endpoint is intentionally limited to inbox watermark checks, rename and document that narrower contract.

**Review by astaxie — COMMENTED** [review]
## Review summary / 审查总结

**中文**

本次审查发现 3 个阻塞问题和 3 个需要补齐或澄清的需求问题。下方已添加 5 条行内评论；另有 1 个无法锚定到 diff 的阻塞问题：

- **[P1] 原生安装与升级会启动失败。** `backend/internal/server/config.go` 现在要求生产环境必须提供强 `TOKENHUB_INTEGRATION_TOKEN`，但 `deploy/native/install.sh` 的 `write_initial_config` 没有生成或写入该变量。新安装会回退到 `dev_integration_token` 并被启动校验拒绝；升级会保留缺少该变量的旧配置，结果相同。请为新安装生成独立 Token，为已有配置增加升级迁移，并在 `deploy/native/install_test.sh` 覆盖这两条路径。

验证结果：`go test ./...`、`go vet ./...`、Gateway 定向测试和 `git diff --check` 均通过；项目覆盖、跨租户 usage 关联和过期后幂等重放问题均已通过未提交的最小测试复现。

**English**

This review found three blocking issues and three requirements that need implementation or clarification. Five findings are attached inline. One blocking issue cannot be anchored to the diff:

- **[P1] Native installs and upgrades will fail to start.** `backend/internal/server/config.go` now requires a strong `TOKENHUB_INTEGRATION_TOKEN` in production, but `write_initial_config` in `deploy/native/install.sh` neither generates nor writes it. Fresh installs fall back to `dev_integration_token` and fail startup validation; upgrades preserve older config files without the variable and fail the same way. Please generate a separate token for fresh installs, migrate existing configs during upgrade, and cover both paths in `deploy/native/install_test.sh`.

Validation: `go test ./...`, `go vet ./...`, focused Gateway tests, and `git diff --check` pass. The managed-project overwrite, cross-tenant usage join, and post-expiry idempotency issues were also reproduced with uncommitted minimal tests.

This is a COMMENT review and does not change the PR approval state.

File: backend/internal/server/store.go (hunk 1 above)
Comment (desertsurge) [reply backend/internal/server/store.go]: 已在 `cb78ae0` 修复，并在 `e80d187` 调整 SQLite 锁顺序后通过 CI。`CreateProjectChecked` 会在 `OnConflict UpdateAll` 前拒绝对应 `GatewayProject` 的调用方 ID；项目更新/删除以及团队新增、角色更新、移除也统一受只读保护。`TestGatewayManagedProjectsAndKeysAreReadOnlyToLocalManagement` 覆盖了 POST 覆盖和全部团队写入口。

File: backend/internal/server/gateway_request_logs.go (hunk 2 above)
Comment (desertsurge) [reply backend/internal/server/gateway_request_logs.go]: 已在 `cb78ae0` 修复。usage join 现在同时约束 `request_id` 和 `api_key_id`，并为 `UsageRecord(request_id, api_key_id)` 增加复合索引。`TestGatewayModelAccessKeyLifecycleIsTenantScopedAndIdempotent` 增加了另一租户相同 Request ID 的 usage，验证不会污染当前租户汇总。

File: backend/internal/server/gateway_model_access_keys.go (hunk 3 above)
Comment (desertsurge) [reply backend/internal/server/gateway_model_access_keys.go:172]: 已在 `cb78ae0` 修复。创建输入先做稳定规范化和 digest，查询已有 `control_request_id` 后，仅对真正的新请求执行“过期时间晚于当前时间”的动态校验。`TestGatewayModelAccessKeyReplayRemainsIdempotentAfterExpiry` 验证 Key 过期后完全相同请求仍返回已有记录。

File: backend/internal/server/gateway_model_access_keys.go (hunk 4 above)
Comment (desertsurge) [reply backend/internal/server/gateway_model_access_keys.go:686]: 已在 `cb78ae0` 将组织定义为授权生命周期边界。项目关联组织时，创建和运行时授权都会校验组织状态；用户 Key 还要求 active 的组织成员关系，workload Key 通过项目继承组织状态。组织停用会临时禁用适用 Key，组织删除会永久吊销其项目下的 Key。`TestGatewayManagedUserKeyFollowsOrganizationLifecycle` 覆盖停用、恢复、成员移除/恢复和删除。

File: backend/internal/server/gateway_integration.go (hunk 5 above)
Comment (desertsurge) [reply backend/internal/server/gateway_integration.go:327]: 已在 `cb78ae0` 增加 `projection_summaries`。tenant、organization、tenant_member、organization_member、project、workload 每类均返回 `count`、`max_version` 和稳定 SHA-256 `digest`；记录按 external ID 排序，摘要覆盖关系、名称、状态、版本和删除标记。对账测试验证了摘要结构，并验证组织状态/版本变化会改变 digest 和版本水位；三语部署文档已同步。

**Comment by desertsurge** [comment]
本轮审查问题已在 `cb78ae0` 和 `e80d187` 全部修复。

无法锚定的原生安装问题也已处理：新安装会生成独立的 32-byte `TOKENHUB_INTEGRATION_TOKEN`，升级旧配置时仅在缺失时补写，后续升级不会覆盖已有值，也不会复用 Admin Token。`deploy/native/install_test.sh` 已覆盖新安装、旧配置迁移和保留已有 Token。

验证结果：当前提交 `e80d187` 的 Backend、Frontend、Deployment 三项 GitHub Actions 均通过；本地安装器集成测试、TypeScript 编译、Next.js 生产构建和 `git diff --check` 通过。

**Comment by astaxie** [comment]
因为老文件进行了大量的重构，需要解决冲突问题

File: backend/internal/server/gateway_model_access_keys.go [hunk 6]
```diff
@@ -0,0 +1,740 @@
...
+		}
+		return GatewayModelAccessKeyCreateResult{}, err
+	}
+	if !created {
+		return GatewayModelAccessKeyCreateResult{Key: publicKey(key), Created: false}, nil
+	}
+	return GatewayModelAccessKeyCreateResult{Key: publicKey(key), Secret: rawSecret, Created: true}, nil
+}
+
+const gatewayModelAccessKeyEffectiveStatusExpression = `(CASE
```
Comment (astaxie) [inline backend/internal/server/gateway_model_access_keys.go:271]: [P2] 将组织生命周期纳入列表的有效状态

创建与运行时认证现在会检查项目所属组织以及用户的组织成员关系，但这里的 SQL 状态表达式和下面的批量状态计算仍只检查 tenant、project、principal/workload。组织停用或成员被移除后，Key 已经无法认证，却仍会在列表中返回为 `active`；`status=active` 会错误包含它，`status=disabled` 会错误遗漏它，total 和分页也会随之不准确。我用最小回归测试复现了这一点。请让 SQL 筛选和批量 hydrator 使用与运行时认证相同的组织/成员规则，并补充列表及状态筛选测试。

File: deploy/install.sh [hunk 7]
```diff
@@ -156,6 +157,7 @@ while IFS= read -r line; do
   case "$line" in
     TOKENHUB_ENV=*) tokenhub_environment="${line#*=}" ;;
     TOKENHUB_ADMIN_TOKEN=*) admin_token="${line#*=}" ;;
+    TOKENHUB_INTEGRATION_TOKEN=*) integration_token="${line#*=}" ;;
```
Comment (astaxie) [inline deploy/install.sh:160]: [P1] 在 Compose JSON fallback 中保留 Integration Token

这里新增的解析分支依赖 `compose_environment` 包含 `TOKENHUB_INTEGRATION_TOKEN`，但当 `docker compose config --environment` 不可用、脚本转而使用 `config --format json` 时，上面的 Python fallback 变量列表没有输出该变量。即使 JSON 中已有正确的强 Token，后续也会回退成 `change-me-tokenhub-integration-token` 并被生产校验拒绝，导致该兼容路径下的 `--check-only` 和安装始终失败。请把 Integration Token 加入 fallback 列表，并增加“`--environment` 失败但 JSON 成功”的安装脚本测试。

**Review by astaxie — COMMENTED** [review]
复审确认原先 6 项问题中 5 项已完整修复；组织生命周期仍有一处列表状态不一致，另外 Compose JSON fallback 还存在 Integration Token 丢失问题。下面两条需要继续修复。本 review 仅提交评论，不改变 PR 的审批状态。

File: backend/internal/server/gateway_model_access_keys.go [hunk 8]
```diff
@@ -0,0 +1,740 @@
...
+		ControlRequestDigest: digest,
+		Name:                 normalized.Name,
+		Group:                normalized.Group,
+		KeyHash:              HashSecret(rawSecret),
+		KeyCiphertext:        keyCiphertext,
+		KeyPrefix:            prefix,
+		KeySuffix:            suffix,
+		Allowed:              normalized.AllowedModels,
+		IPAllowlist:          normalized.IPAllowlist,
+		Limits:               normalized.Limits,
```
Comment (astaxie) [inline backend/internal/server/gateway_model_access_keys.go:214]: [P1] 将分钟限额写入当前准入逻辑读取的字段

Integration API 接受 `limits.rate_limit_rpm` 和 `limits.token_limit_tpm`，但这里仅把它们写入嵌入的 `APIKey.Limits`。当前请求准入路径在 `store_routing_calls.go` 中会先把这两个嵌入字段清零，再只读取独立的 `APIKey.RateLimitRPM` / `TokenLimitTPM` 指针，因此 AIP 创建 `limits: {"rate_limit_rpm": 1}` 的 Key 实际仍可无限请求。请把这两个值映射到当前规范字段（并明确 0/未设置语义）、校验非负值，并补充第二个分钟内请求被拒绝的回归测试。

File: backend/internal/server/gateway_integration.go [hunk 9]
```diff
@@ -0,0 +1,897 @@
...
+	item.OrganizationID, item.OwnerPrincipalID = organizationID, ownerID
+	if name := payloadString(event.Payload, "name"); name != "" {
+		item.Name = name
+	}
+	item.Status, item.DeletedAt = projectionStatus(event, item.Status, item.DeletedAt, now)
+	item.Version, item.SyncedAt = event.Version, now
+	if err := tx.Save(&item).Error; err != nil {
+		return "", "project", 0, "", err
+	}
+	if err := syncGatewayServingProject(tx, item, now); err != nil {
```
Comment (astaxie) [inline backend/internal/server/gateway_integration.go:805]: [P1] 删除项目时永久吊销已有 Key

`project.archived` 和 `project.deleted` 目前都只会把 serving Project 设为 disabled；这里没有像 tenant、organization、workload 删除那样吊销项目下的 Key。之后若同一 external project 收到更高版本的 `project.updated` 且状态恢复 active，`projectionStatus` 会清除 `DeletedAt`，旧 Key 也会重新通过运行时 scope 校验。Archive 已经提供可恢复语义，Delete 不应让已删除项目的旧凭据在重建后复活。请仅在 `project.deleted` 时永久 revoke 项目下的托管 Key，并覆盖 delete → recreate 后旧 secret 仍不可用的测试。

File: backend/internal/server/gateway_usage.go [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+	}
+	for _, item := range logDaily {
+		current := daily[item.Date]
+		current.Date, current.RequestCount, current.ErrorCount = item.Date, item.RequestCount, item.ErrorCount
+		daily[item.Date] = current
+	}
+	offset := time.Duration(filter.TimezoneOffsetMinutes) * time.Minute
+	dateFrom := filter.DateFrom.Add(offset)
+	dateTo := filter.DateTo.Add(offset)
+	for date := dateFrom; !date.After(dateTo); date = date.AddDate(0, 0, 1) {
```
Comment (astaxie) [inline backend/internal/server/gateway_usage.go:195]: [P2] 按日历日期而不是具体时刻补齐 timeseries

这里从 `date_from` 的具体时刻开始逐日递增，并用完整时间与 `date_to` 比较。例如查询 7/27 12:00 到 7/28 10:00 时，SQL 汇总和 breakdown 会包含 7/28 上午的数据，但循环下一步是 7/28 12:00，因超过结束时间而完全漏掉 7/28 的 timeseries bucket。请先在请求时区偏移下把两端归一化为日历日期，再按日期闭区间迭代，并增加跨天但非整日的回归测试。

File: backend/internal/server/config.go [hunk 11]
```diff
@@ -183,10 +185,13 @@ func (c Config) ValidateForStartup() error {
 	case "dev", "development", "local", "test":
 		return nil
 	}
-	invalid := make([]string, 0, 3)
+	invalid := make([]string, 0, 4)
 	if reason := weakProductionSecretReason(c.AdminToken, 32, "dev_admin_token", "change-me-tokenhub-admin-token"); reason != "" {
 		invalid = append(invalid, "TOKENHUB_ADMIN_TOKEN "+reason)
 	}
+	if reason := weakProductionSecretReason(c.IntegrationToken, 32, "dev_integration_token", "change-me-tokenhub-integration-token"); reason != "" {
```
Comment (astaxie) [inline backend/internal/server/config.go:192]: [P2] 在生产配置中拒绝复用 Admin Token

PR 和部署文档都把 `TOKENHUB_INTEGRATION_TOKEN` 定义为专用凭证，并明确不能由 Admin Token 替代；但启动校验只分别检查长度和占位值，同一个 32 字节字符串同时配置给 Admin 与 Integration 仍会通过。这样 Admin Token 就可以调用拓扑写入以及 Key reveal/revoke 接口，实际消除了两个权限面的隔离。请在后端启动校验和部署脚本校验中拒绝两个 Token 相等，并补对应测试。

**Review by astaxie — COMMENTED** [review]
本轮复审发现 4 个新的实现问题，已分别添加行内评论。此前两条未解决评论（组织生命周期未进入 Key 列表有效状态、Compose JSON fallback 丢失 Integration Token）在当前 `c94ee63` 中仍然有效，因此没有重复留言。

验证：`go test ./...`、`go vet ./...`、两套安装器测试、Compose 渲染、repository gates 和 `git diff --check` 均通过；GitHub CI 也为绿色。本 review 仅提交评论，不改变 PR 审批状态。

**Review by astaxie — COMMENTED** [review]
Re-review of current head 1da32c4: this is not merge-ready. The only changes since reviewed head c94ee63 are the merge from upstream main; none of the six files called out in the previous review changed. The organization lifecycle list status, Compose JSON fallback token, canonical RPM/TPM fields, permanent project-delete revocation, calendar-date timeseries fill, and Admin/Integration token separation findings therefore all remain actionable. Green CI confirms the merged tree builds, but does not resolve those behavior gaps.

File: backend/internal/server/store_database.go [hunk 12]
```diff
@@ -952,6 +952,13 @@ func schemaModels() []any {
 	return []any{
 		&billingpersistence.ConnectorRow{}, &billingpersistence.RecordRow{}, &billingpersistence.RawSnapshotRow{}, &billingpersistence.SyncRunRow{},
 		&ReconciliationRule{}, &ReconciliationRun{}, &ReconciliationItem{},
+		&GatewayTenant{},
```
Comment (astaxie) [inline backend/internal/server/store_database.go:955]: [P1] Add a forward migration for existing databases

`schemaModels()` is the frozen bridge-release model set, and the baseline SQL is applied only to fresh/adopted databases. Adding the Gateway tables and new API-key fields here and regenerating `000001_baseline.json` does not update installations whose ledger already contains the current baseline and migrations through v6; those databases never rerun legacy AutoMigrate, so the tables/columns remain absent and semantic verification or the new endpoints fail after upgrade. Add an expand migration for both SQLite and PostgreSQL, keep the frozen baseline contract intact, and test upgrading a current-main v6 database while preserving existing rows.

File: backend/internal/server/gateway_integration_test.go [hunk 13]
```diff
@@ -0,0 +1,1047 @@
...
+func TestGatewayIntegrationEndpointRequiresDedicatedToken(t *testing.T) {
+	store := NewMemoryStore()
+	app := NewWithConfig(store, Config{
+		AdminToken:       "admin_token",
+		IntegrationToken: "integration_token",
+		SecretKey:        "test_secret",
+	}).Handler()
+	event := gatewayIntegrationEvent("evt_auth", "tenant.created", "tenant", "tenant_01", "tenant_01", 1, map[string]interface{}{
+		"externalId": "tenant_01",
+		"name":       "企业一",
```
Comment (astaxie) [inline backend/internal/server/gateway_integration_test.go:37]: [P2] Keep backend fixtures in the repository source language

This new backend integration test uses Chinese developer-facing fixture values here and repeatedly throughout the file (`企业一`, `研发中心`, `最新名称`, `智能客服`, and others). AGENTS.md requires source fixtures and test data to be English; the localized-selector exception applies only to tests intentionally exercising translated UI. Replace these fixtures and their assertions with English equivalents, then run gofmt and the affected Gateway tests.

**Review by astaxie — CHANGES_REQUESTED** [review]
Re-reviewed the complete current diff after the upstream merge. The six existing unresolved findings remain; two additional issues are attached inline. Focused Gateway tests, all 175 repository tests, translation/UI/environment/source-line gates, Compose rendering, and git diff --check pass locally, and current GitHub checks are green. The missing forward migration blocks upgrades from existing databases, so this head is not ready for approval.
